src/librand/distributions/mod.rs

```diff
@@ -103,7 +107,8 @@ impl<'a, T: Clone> WeightedChoice<'a, T> {
     /// - the total weight is larger than a `usize` can contain.
     pub fn new(items: &'a mut [Weighted<T>]) -> WeightedChoice<'a, T> {
         // strictly speaking, this is subsumed by the total weight == 0 case
-        assert!(!items.is_empty(), "WeightedChoice::new called with no items");
+        assert!(!items.is_empty(),
```

Comment:
We split when the internals of a function call or function-like macro is more than a heuristic limit (I think 60 chars). Mostly this works well. Occasionally (like here) it seems a bit over the top. I expect its a heuristic we'll polish as time goes by.